### DISCUSSION TITLE
Quill Inline styles

### DISCUSSION THREAD

**Discussion by michaeljcoyne** [area17/twill#1284] [General]
Is there away to enable inline styles via config for quill?

https://quilljs.com/guides/how-to-customize-quill/#class-vs-inline

**Comment by pboivin** [comment]
Hey @michaeljcoyne, you can customize the [built-in configuration for Quill](https://github.com/area17/twill/blob/2.x/frontend/js/libs/Quill/QuillConfiguration.js).

This can be done as part of a custom build script, before running the `twill:build` command, for example :
```sh
cp resources/assets/js/libs/Quill/QuillConfiguration.js vendor/area17/twill/frontend/js/libs/Quill/

php artisan twill:build
```

This workflow will surely be improved in the near future.

**Comment by michaeljcoyne** [comment]
Hi Patrick, thank you.  I tried that, putting in my  "twill-build" script and running php artisan twill:build. However the file is not copied across. Any Ideas why?

```
"scripts": {
        "dev": "npm run development",
        "development": "mix",
        "watch": "mix watch",
        "watch-poll": "mix watch -- --watch-options-poll=1000",
        "hot": "mix watch --hot",
        "prod": "npm run production",
        "production": "mix --production",
        "twill-build": "cp resources/assets/js/libs/Quill/QuillConfiguration.js vendor/area17/twill/frontend/js/libs/Quill/QuillConfiguration.js && npm run twill-copy-blocks && cd vendor/area17/twill && npm ci && npm run prod && cp -R public/* ${INIT_CWD}/public",
        "twill-copy-blocks": "npm run twill-clean-blocks && mkdir -p resources/assets/js/blocks/ && mkdir -p vendor/area17/twill/frontend/js/components/blocks/customs/ && cp -R resources/assets/js/blocks/ vendor/area17/twill/frontend/js/components/blocks/customs",
        "twill-clean-blocks": "rm -rf vendor/area17/twill/frontend/js/components/blocks/customs/*"
    },
```

**Reply by pboivin** [reply]
Yes, the `twill-build` and `twill-*-blocks` NPM scripts are not needed on recent versions of Twill. Running `php artisan twill:build` will take care of doing all the required steps.

If you want to dig a bit deeper into the frontend build/dev workflow, you can read part 1 of this short tutorial:
https://gist.github.com/pboivin/19d49cbb3c0a6c5c06469117597cc0f7

So in your case, you will only need to run `php artisan twill:build` when you customize `QuillConfiguration.js` and when you update Twill via composer.

What I usually do for this kind of override is to wrap it in a small shell script, something like :

**file: build.sh**
```
#!/usr/bin/env bash

cp resources/assets/js/libs/Quill/QuillConfiguration.js vendor/area17/twill/frontend/js/libs/Quill/

php artisan twill:build
```

Then I simply run `bash build.sh` to trigger the build.

Hope this can help!

**Comment by michaeljcoyne** [comment]
Hi Patrick so basically you are saying to override vendor files? And run that bash each time I do an update? As all that command does is save you typing it.

I would have thought hooking in via the build command or other php override like event, or just a config file.

Might be easier to just build a custom block and choose my editor which looks possible. 

Thanks.
